### PR TITLE
Add spec directly for UsernameValidator.perform_validation

### PR DESCRIPTION
**Pull request by compwron**
I didn't see any existing direct tests and I was looking for a small thing to do. https://github.com/discourse/discourse/search?utf8=%E2%9C%93&q=perform_validation

### CODE REVIEW

**Comment by compwron** [comment]
I am relatively sure that I didn't break the failing test and it passes on my machine... halp

````
1) PrettyText Cooking off topic quoting can correctly populate topic title
     Failure/Error: expect(PrettyText.cook("[quote=\"EvilTrout, post:2, topic:#{topic.id}\"]ddd\n[/quote]", topic_id: 1)).to match_html expected
       after sanitizing for extra white space and compactness, expected:
       <aside class="quote" data-post="2" data-topic="1"><div class="title">
       <div class="quote-controls"></div>EvilTrout:</div>
       <blockquote><p>ddd</p></blockquote></aside>
        to match:
       <aside class="quote" data-post="2" data-topic="1"><div class="title">
       <div class="quote-controls"></div><a href="http://test.localhost/t/this-is-a-test-topic/1/2">This is a test topic</a>
       </div>
       <blockquote><p>ddd</p></blockquote></aside>
     # ./spec/components/pretty_text_spec.rb:20:in `block (4 levels) in <top (required)>'
Finished in 10 minutes 3 seconds (files took 10.1 seconds to load)
5053 examples, 1 failure, 9 pending
````

**Comment by tgxworld** [comment]
@compwron  Can you try rebasing this again master? Thanks!

File: spec/models/username_validator_spec.rb [hunk 1]
```diff
@@ -14,4 +14,27 @@
...
+      it 'does not add errors to the passed in user' do
+        expect(user.errors.messages).to eq({:email=>[]})
+      end
+    end
+
+    context "with an invalid username" do
+      let(:user) { Fabricate(:user, name: 'invalid name') }
+      it 'adds errors to the passed in user' do
+        expect(subject).to eq ["must only include numbers, letters and underscores"]
+        expect(user.errors.messages).to eq({:email=>[], :name=>["must only include numbers, letters and underscores"]})
```
Comment (tgxworld) [inline spec/models/username_validator_spec.rb]: nitpick: we prefer `email: []` instead of `:email => []`

**Comment by SamSaffron** [comment]
I am so sorry, I am closing this as too minor, #valid_format? tests above already take care of this as a side effect, this test seems very specific and I am not sure we need to carry it around.

**Pull request closed without merging**